src/main.rs

```diff
@@ -183,6 +190,42 @@ impl RequestHandler for Handler {
 	}
 }
 
+async fn load_cert_and_key(
+	cert_path: PathBuf,
+	key_path: PathBuf
+) -> anyhow::Result<(Vec<Certificate>, PrivateKey)> {
+	let certificates: Vec<_> = rustls_pemfile::read_all(&mut BufReader::new(
+		File::open(&cert_path)
+			.with_context(|| format!("failed to open {:?}", cert_path))?
+	))
+	.with_context(|| format!("failed to prase {:?}", cert_path))?
+	.iter()
+	.filter_map(|cert| match cert {
+		rustls_pemfile::Item::X509Certificate(cert) => Some(Certificate(cert.to_owned())),
+		_ => None
+	})
+	.collect();
+	if certificates.is_empty() {
+		bail!(format!(
+			"no DER-encoded x509 certificate found in {:?}",
```

Comment:
This is technically incorrect. `rustls-pemfile` will in fact not read DER-encoded files, but rather PEM encoded files and convert the items to their respective DER encoding.